src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -314,7 +314,7 @@ public static bool IsUnnecessaryCast(this CastExpressionSyntax cast, SemanticMod
 
             if (expressionType != null 
                 && expressionType.IsEnumType() 
-                && (cast.IsParentKind(SyntaxKind.UnaryMinusExpression) || cast.IsParentKind(SyntaxKind.UnaryPlusExpression)))
+                && (cast.WalkUpParentheses().IsParentKind(SyntaxKind.UnaryMinusExpression) || cast.WalkUpParentheses().IsParentKind(SyntaxKind.UnaryPlusExpression)))
```

Comment:
:bulb: There is an overload of `IsParentKind` which accepts two `SyntaxKind` arguments. Using this will allow you to call `WalkUpParentheses` one time instead of two.

:bulb: Maybe extract this block into a method `EnumCastDefinitelyCantBeRemoved`, similar to the existing `PointerCastDefinitelyCantBeRemoved` method?